packages/catalog-client/src/CatalogClient.ts

```diff
@@ -140,39 +167,50 @@ export class CatalogClient implements CatalogApi {
       .find(l => locationCompound === stringifyLocationReference(l));
   }
 
+  async removeLocationById(
+    id: string,
+    options?: CatalogRequestOptions,
+  ): Promise<void> {
+    await this.requestIgnored('DELETE', `/locations/${id}`, options);
+  }
+
   async removeEntityByUid(
     uid: string,
     options?: CatalogRequestOptions,
   ): Promise<void> {
-    const response = await fetch(
-      `${await this.discoveryApi.getBaseUrl('catalog')}/entities/by-uid/${uid}`,
-      {
-        headers: options?.token
-          ? { Authorization: `Bearer ${options.token}` }
-          : {},
-        method: 'DELETE',
-      },
-    );
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-    return undefined;
+    await this.requestIgnored('DELETE', `/entities/by-uid/${uid}`, options);
   }
 
   //
   // Private methods
   //
 
-  private async getRequired(
+  private async requestIgnored(
+    method: string,
+    path: string,
+    options?: CatalogRequestOptions,
+  ): Promise<void> {
+    const url = `${await this.discoveryApi.getBaseUrl('catalog')}${path}`;
+    const headers = new Headers(
+      options?.token ? { Authorization: `Bearer ${options.token}` } : {},
+    );
```

Comment:
With this change, I get an exception in the backend. Seems like `Headers` is not usable in nodejs?

```
[1] 2021-03-30T09:13:58.265Z backstage error Headers is not defined type=errorHandler stack=ReferenceError: Headers is not defined
[1]     at CatalogClient.requestRequired (/Users/dominikhenneke/git/terminal/node_modules/@backstage/catalog-client/dist/index.cjs.js:101:21)
[1]     at async CatalogClient.getEntities (/Users/dominikhenneke/git/terminal/node_modules/@backstage/catalog-client/dist/index.cjs.js:36:22)
[1]     at async CatalogTransformer.getAnnotatedEntities (webpack-internal:///../../plugins/catalog-federation-backend/src/transform/CatalogTransformer.ts:97:22)
[1]     at async CatalogTransformer.transform (webpack-internal:///../../plugins/catalog-federation-backend/src/transform/CatalogTransformer.ts:44:34)
[1]     at async Cached.get (webpack-internal:///../backend-common/src/cache/cached.ts:26:14)
[1]     at async CacheMap.get (webpack-internal:///../backend-common/src/cache/cachemap.ts:47:12)
[1]     at async eval (webpack-internal:///../../plugins/catalog-federation-backend/src/service/router.ts:81:38)
```

Reply:
Oh no! Hm maybe we can just remove that usage? @dhenneke can you check on your end (since you have it being exercised) that just `const headers = options?.token ? { Authorization: `Bearer ${options.token}` } : {};` works out to use, essentially, instead? And maybe make the PR

Reply:
Otherwise let me know and I'll try to fix it immediately